src/powershell/FinOpsToolkit.psm1

```diff
@@ -0,0 +1,306 @@
+        Azure location to execute the deployment from.
+
+    .PARAMETER Version
+        Defaults to latest. Version of FinOps hub template to use.
+
+    .PARAMETER StorageSku
+        Optional. Storage account SKU. Premium_LRS = Lowest cost, Premium_ZRS = High availability. Note Standard SKUs are not available for Data Lake gen2 storage.
+
+    .PARAMETER Tags
+        Optional. Tags for all resources.
+
+    .PARAMETER ExportScopes
+        Optional. List of scope IDs to create exports for.
+
+    .EXAMPLE
+        Deploy-FinOpsHub -HubName MyHub -ResourceGroup MyExistingResourceGroup -Location westus
+        
+        Deploys a new FinOps hub instance named MyHub to an existing resource group name MyExistingResourceGroup.
+    
+    .EXAMPLE
+        Deploy-FinOpsHub -HubName MyHub -Location westus -Version 1.0.0
+        
+        Deploys a new FinOps hub instance named MyHub using version 1.0.0 of the template.
+#>
+function Deploy-FinOpsHub
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]
+        $HubName,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $ResourceGroup,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Location,
+
+        [Parameter()]
+        [string]
+        $Version = 'latest',
+
+        [Parameter()]
+        [ValidateSet('Premium_LRS', 'Premium_ZRS')]
+        [string]
+        $StorageSku = 'Premium_LRS',
+
+        [Parameter()]
+        [hashtable]
+        $Tags,
+
+        [Parameter()]
+        [array]
+        $ExportScopes
+    )
+
+    $toolkitPath = Join-Path -Path $env:temp -ChildPath 'FinOpsHub'
+    $availableVersions = Get-FinOpsToolkitVersions
+    if ($Version -ne 'latest' -and $availableVersions -notcontains $Version)
+    {
+        throw ($LocalizedData.VersionNotFound -f $Version)
+    }
+
+    Save-GitHubRelease -RepositoryName 'microsoft/cloud-hubs' -Tag $Version -Destination $toolkitPath
+    $toolkitFile = Get-ChildItem -Path $toolkitPath -Include 'finops-hub/main.bicep' -Recurse
+    $parameterSplat = @{
+        TemplateFile            = $toolkitFile.FullName
+        TemplateParameterObject = @{
+            HubName = $HubName
+        }
+    }
+
+    foreach ($parameter in @('StorageSku', 'Tags', 'ExportScopes'))
+    {
+        if ($PSBoundParameters.ContainsKey($parameter))
+        {
+            $parameterSplat.TemplateParameterObject.Add($parameter, $PSBoundParameters[$parameter])
+        }
+    }
+
+    try
+    {
+        $resourceGroup = Get-AzResourceGroup -Name $ResourceGroup -Location $Location -ErrorAction 'Stop'
+    }
+    catch
+    {
+        $resourceGroup = New-AzResourceGroup -Name $ResourceGroup -Location $Location
+    }
+    
+    try
+    {
+        Write-Verbose -Message ($LocalizedData.DeployFinOpsHub -f $Path, $resourceGroup.ResourceGroupName)
+        $deployment = New-AzResourceGroupDeployment @parameterSplat -ResourceGroupName $resourceGroup.ResourceGroupName
```

Comment:
I think we're missing `-TemplateFile` and I believe the parameters would need `-TemplateParameterObject`. Also, shouldn't that be prefixed with `$` and not `@`?